api/src/opentrons/protocol_engine/types.py

```diff
@@ -967,7 +967,16 @@ class EnumParameter(RTPBase):
     )
 
 
-RunTimeParameter = Union[NumberParameter, EnumParameter, BooleanParameter]
+class CSVParameter(RTPBase):
+    """A CSV file parameter defined in a protocol."""
+
+    id: Optional[str] = Field(
+        ...,
+        description="The UUID identifier of the file stored on the robot. For local analysis this will be empty.",
+    )
```

Comment:
From the in-person chat: we probably want this to be something like-

```suggestion
    file: Optional[FileId] = Field(
        ...,
        description="The CSV file stored on the robot, to be used as the CSV RTP override value. For local analysis this will be empty.",
    )
```
where `FileId` is:

```
class FileId(BaseModel):
    """A CSV file UUID descriptor."""

    id: Optional[str] = Field(
        ...,
        description="The UUID identifier of the file stored on the robot. For local analysis this will be empty.",
    )
```

Reply:
Also, let's add the `type` as `Literal["csv_file"]`